pkg/apis/certmanager/validation/issuer_test.go

```diff
@@ -146,6 +147,19 @@ func TestValidateACMEIssuerConfig(t *testing.T) {
 				HTTP01:     &v1alpha1.ACMEIssuerHTTP01Config{},
 			},
 		},
+		"acme issue with invalid http01 service config": {
+			spec: &v1alpha1.ACMEIssuer{
+				Email:      "valid-email",
+				Server:     "valid-server",
+				PrivateKey: validSecretKeyRef,
+				HTTP01: &v1alpha1.ACMEIssuerHTTP01Config{
+					SolverServiceType: corev1.ServiceType("InvalidServiceType"),
+				},
+			},
+			errs: []*field.Error{
+				field.Invalid(fldPath.Child("http01", "solverServiceType"), corev1.ServiceType("InvalidServiceType"), "optional field solverServiceType must be one of [\"ClusterIP\" \"NodePort\"]"),
+			},
+		},
```

Comment:
Could you add test cases for the *valid* types too? Sorry to be a pain! But for completeness 😄